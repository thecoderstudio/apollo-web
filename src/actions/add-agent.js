const SELECT_OPERATING_SYSTEM = 'SELECT_OPERATING_SYSTEM';
const SELECT_ARCHITECTURE = 'SELECT_ARCHITECTURE';

function selectOperatingSystem(operatingSystem) {
  return {
    type: SELECT_OPERATING_SYSTEM,
    selectedOperatingSystem: operatingSystem
  };
}

function selectArchitecture(architecture) {
  return {
    type: SELECT_ARCHITECTURE,
    selectedArchitecture: architecture
  };
}

export {
  selectOperatingSystem,
  SELECT_OPERATING_SYSTEM,
  selectArchitecture,
  SELECT_ARCHITECTURE
};
