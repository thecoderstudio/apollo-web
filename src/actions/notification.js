var id = 0

function notify(message, severity) {
  return {
    type: 'NOTIFY',
    id: id++,
    message,
    severity
  }
}

function dismiss(notificationId) {
  return {
    type: 'DISMISS',
    id: notificationId
  }
}

export { notify, dismiss };
