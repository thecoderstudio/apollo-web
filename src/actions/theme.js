function changeTheme(theme) {
  return {
    type: 'CHANGE_THEME',
    theme
  };
};

export { changeTheme };
