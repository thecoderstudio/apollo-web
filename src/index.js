import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'
import { render } from 'react-dom';
import App from './App';
import { store, persistor } from './store';

render(
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <App />
    </PersistGate>
  </Provider>,
  document.getElementById('app')
);
