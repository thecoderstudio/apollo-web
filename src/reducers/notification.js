import { Map } from 'immutable';

export default function notificationReducer(state = Map({}), action) {
  switch (action.type) {
    case 'NOTIFY':
      return state.set(action.id, {
        message: action.message,
        severity: action.severity
      });
    case 'DISMISS':
      return state.delete(action.id);
    default:
      return state;
  }
}
