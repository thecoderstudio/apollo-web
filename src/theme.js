export const darkTheme = {
  name: "dark",
  black: "#181818",
  lightBlack: "#1B1B1B",
  white: "#F8F8F8",
  primary: "#02A3EE",
  accent: "#4D02EE",
  connectedColor: "#27AE60",
  disconnectedColor: "#C0392B",
  connectingColor: "#F39C12",
  error: "#ED3E2A"
};

function getBrowserPreferredTheme() {
  return darkTheme;
}

export const browserPreferredTheme = getBrowserPreferredTheme();
