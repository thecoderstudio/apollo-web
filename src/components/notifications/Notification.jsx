import React from 'react';
import { connect } from 'react-redux';
import styled, { withTheme } from 'styled-components';
import PropTypes from 'prop-types';
import { severity, dismiss } from '../../actions/notification';

const propTypes = {
  message: PropTypes.string.isRequired,
  id: PropTypes.number.isRequired,
  severity: PropTypes.string,
  timeToLiveInSeconds: PropTypes.number
};

const defaultProps = {
  severity: severity.info,
  timeToLiveInSeconds: null
};

const Container = styled.div`
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: ${props => props.color};
  box-shadow: 0 1px 1px rgba(0,0,0,0.12),
              0 2px 2px rgba(0,0,0,0.12),
              0 4px 4px rgba(0,0,0,0.12),
              0 8px 8px rgba(0,0,0,0.12),
              0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
`;

const CloseIcon = styled.i`
  float: right;
  color: white;
`;

const Message = styled.h5`
  color: white;
`;

class Notification extends React.PureComponent {
  constructor(props) {
    super(props);
    this.getColor = this.getColor.bind(this);
    this.dismissSelf = this.dismissSelf.bind(this);
    this.scheduleDismiss = this.scheduleDismiss.bind(this);
    if (props.timeToLiveInSeconds !== null) {
      this.scheduleDismiss(props.timeToLiveInSeconds);
    }
  }

  componentDidUpdate(prevProps) {
    if (prevProps.timeToLiveInSeconds === this.props.timeToLiveInSeconds) {
      return;
    }

    clearTimeout(this.timeout);

    if (this.props.timeToLiveInSeconds === null) {
      return;
    }

    this.scheduleDismiss(this.props.timeToLiveInSeconds);
  }

  getColor(notificationSeverity) {
    switch(notificationSeverity) {
      case severity.warning:
        return this.props.theme.warning;
      case severity.error:
        return this.props.theme.error;
      default:
        return this.props.theme.accent;
    }
  }

  dismissSelf() {
    this.props.dispatch(dismiss(this.props.id));
  }

  scheduleDismiss(timeToLiveInSeconds) {
    this.timeout = setTimeout(this.dismissSelf, timeToLiveInSeconds * 1000);
  }

  render() {
    let color = this.getColor(this.props.severity);

    return (
      <Container className={this.props.className} color={color}>
        <CloseIcon className="fas fa-times" onClick={this.dismissSelf}/>
        <Message>{this.props.message}</Message>
      </Container>
    );
  }
}

Notification.propTypes = propTypes;
Notification.defaultProps = defaultProps; 

export default withTheme(connect()(Notification));
