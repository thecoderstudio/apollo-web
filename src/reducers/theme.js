import { browserPreferredTheme } from '../theme';

export default function themeReducer(state = { theme: browserPreferredTheme }, action) {
  switch (action.type) {
    case 'CHANGE_THEME':
      return { theme: action.theme };
    default:
      return state;
  }
}
