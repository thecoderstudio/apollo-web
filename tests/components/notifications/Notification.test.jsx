import React from 'react';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import waitForExpect from 'wait-for-expect';
import { dismiss } from '../../../src/actions/notification';
import { darkTheme } from '../../../src/theme';
import Notification from '../../../src/components/notifications/Notification';

const mockStore = configureStore([]);

function getComponentUnrendered(store, id, message, ttl=null, severity='info') {
  return (
    <Provider store={store}>
      <Notification id={id} message={message} theme={darkTheme} timeToLiveInSeconds={ttl} severity={severity} />
    </Provider>
  );
}

function getComponent(store, id, message, ttl=null, severity='info') {
  return renderer.create(getComponentUnrendered(store, id, message, ttl, severity));
}

describe('notification', () => {
  let store;

  beforeEach(() => {
    store = mockStore({

    });
    store.dispatch = jest.fn();
  });

  it("renders correctly for different severity levels", () => {
    let tree = getComponent(store, 0, 'test').toJSON();
    expect(tree).toMatchSnapshot();

    tree = getComponent(store, 0, 'test', null, 'warning').toJSON();
    expect(tree).toMatchSnapshot();

    tree = getComponent(store, 0, 'test', null, 'error').toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("is dismissable using click", () => {
    const component = getComponent(store, 0, 'test'); 
    const icon = component.root.findByType('i');
    icon.props.onClick();

    expect(store.dispatch).toBeCalledWith(dismiss(0));
  });

  it("dismisses if TTL has expired", async () => {
    const component = getComponent(store, 0, 'test', 1); 
    const tree = component.toJSON();
    await waitForExpect(() => {
      expect(store.dispatch).toHaveBeenCalledWith(dismiss(0));
    });
  });

  it("correctly updates TTL", async () => {
    const component = getComponent(store, 0, 'test', null); 
    let tree = component.toJSON();
    component.update(getComponentUnrendered(store, 0, 'test', 1));
    tree = component.toJSON();
    component.update(getComponentUnrendered(store, 0, 'test', null));
    tree = component.toJSON();

    await waitForExpect(() => {
      expect(store.dispatch).toHaveBeenCalledWith(dismiss(0));
    });
  });
});
