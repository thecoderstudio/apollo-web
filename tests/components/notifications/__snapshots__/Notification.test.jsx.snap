// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification renders correctly for different severity levels 1`] = `
.c0 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #4D02EE;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c1 {
  float: right;
  color: white;
}

.c2 {
  color: white;
}

<div
  className="c0"
  color="#4D02EE"
>
  <i
    className="c1 fas fa-times"
    onClick={[Function]}
  />
  <h5
    className="c2"
  >
    test
  </h5>
</div>
`;

exports[`notification renders correctly for different severity levels 2`] = `
.c0 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #ED852A;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c1 {
  float: right;
  color: white;
}

.c2 {
  color: white;
}

<div
  className="c0"
  color="#ED852A"
>
  <i
    className="c1 fas fa-times"
    onClick={[Function]}
  />
  <h5
    className="c2"
  >
    test
  </h5>
</div>
`;

exports[`notification renders correctly for different severity levels 3`] = `
.c0 {
  max-width: 700px;
  width: 95%;
  padding: 8px 16px;
  margin: 8px;
  border-radius: 5px;
  background-color: #ED3E2A;
  box-shadow: 0 1px 1px rgba(0,0,0,0.12), 0 2px 2px rgba(0,0,0,0.12), 0 4px 4px rgba(0,0,0,0.12), 0 8px 8px rgba(0,0,0,0.12), 0 16px 16px rgba(0,0,0,0.12);
  box-sizing: border-box;
}

.c1 {
  float: right;
  color: white;
}

.c2 {
  color: white;
}

<div
  className="c0"
  color="#ED3E2A"
>
  <i
    className="c1 fas fa-times"
    onClick={[Function]}
  />
  <h5
    className="c2"
  >
    test
  </h5>
</div>
`;
